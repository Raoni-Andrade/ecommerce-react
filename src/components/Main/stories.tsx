import { Meta, StoryObj } from '@storybook/react'

import Main from '.'

export default {
  title: 'Main',
  component: Main,
  parameters: {
    layout: 'fullscreen'
  }
} as Meta

export const Default: StoryObj = {}

export const Basic: StoryObj = {
  args: {
    title: 'Trying to learn Storybook',
    description: "First time using Storybook, let's see how it goes!"
  }
}
