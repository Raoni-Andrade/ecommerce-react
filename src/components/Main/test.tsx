import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('Heading is rendering fine', () => {
    // renderiza o component
    const { container } = render(<Main />)

    // busca o elemento e verifica a existência dele
    expect(
      screen.getByRole('heading', { name: /react avançado/i })
    ).toBeInTheDocument()

    // gerar snapshot
    expect(container.firstChild).toMatchSnapshot()
  })

  it('Render the colors correctly', () => {
    // renderiza o component
    const { container } = render(<Main />)

    // verifica se a cor do background está correta
    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
  })
})
