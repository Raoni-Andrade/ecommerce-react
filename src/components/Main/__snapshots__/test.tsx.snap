// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Heading is rendering fine 1`] = `
<main>
  <h1>
    React Avan√ßado
  </h1>
</main>
`;
